packages/next-auth/src/next/index.ts

```diff
@@ -83,25 +83,31 @@ function NextAuth(
 export default NextAuth
 
 export async function getServerSession(
-  context:
-    | GetServerSidePropsContext
-    | { req: NextApiRequest; res: NextApiResponse },
-  options: NextAuthOptions
+  ...args:
+    | [GetServerSidePropsContext['req'], GetServerSidePropsContext['res'], NextAuthOptions]
+    | [NextApiRequest, NextApiResponse, NextAuthOptions]
 ): Promise<Session | null> {
+  console.warn(
+    "[next-auth][warn][EXPERIMENTAL_API]",
+    "\n`getServerSession` is experimental and may be removed in the future.",
```

Comment:
```suggestion
    "\n`getServerSession` is experimental and may be removed or changed in the future.",
```

https://github.com/nextauthjs/next-auth/pull/4116/files#r819707365

This also needs to be mentioned in https://github.com/nextauthjs/next-auth/blob/main/docs/docs/warnings.md